cmd/lifecycle/exporter.go

```diff
@@ -153,26 +169,27 @@ func (e *exportCmd) Exec() error {
 		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
 	}
 
-	cacheStore, err := initCache(e.cacheImageTag, e.cacheDir)
+	cacheStore, err := initCache(e.cacheImageTag, e.cacheDir, e.keychain)
 	if err != nil {
 		cmd.DefaultLogger.Infof("no stack metadata found at path '%s', stack metadata will not be exported\n", e.stackPath)
 	}
 
 	return e.export(group, cacheStore, analyzedMD)
 }
 
-func (ea exportArgs) export(group lifecycle.BuildpackGroup, cacheStore lifecycle.Cache, analyzedMD lifecycle.AnalyzedMetadata) error {
-	ref, err := name.ParseReference(ea.imageNames[0], name.WeakValidation)
-	if err != nil {
-		return cmd.FailErr(err, "failed to parse registry")
+func (e *exportCmd) registryImages() []string {
+	var registryImages []string
+	if e.cacheImageTag != "" {
+		registryImages = append(registryImages, e.cacheImageTag)
 	}
-	registry := ref.Context().RegistryStr()
-
-	stackMD, runImageRef, err := resolveStack(ea.stackPath, ea.runImageRef, registry)
-	if err != nil {
-		return err
+	if !e.useDaemon {
+		registryImages = append(registryImages, e.imageNames...)
+		registryImages = append(registryImages, e.runImageRef)
```

Comment:
Worth noting that the previous (analyzed) image is also a registry image that we access. However, this currently won't break because we ensure the previous image and the destination tag are on the same registry and our `DefaultKeychain` actually only cares about registries not images.

Reply:
We'll pass in previous now. In the future, we could send in registries instead of images.